types/config.go

```diff
@@ -162,6 +162,39 @@ type ProviderConfig struct {
 	// azure: https://azure.microsoft.com/en-us/global-infrastructure/geographies/#overview
 	// gcp: https://cloud.google.com/compute/docs/regions-zones#available
 	Zone string `cloud:"zone"`
+
+	// AwsIAM is used to define IAM Role AWS
+	AwsIAM *AwsIAM
+}
+
+// AwsIAM define IAM Role AWS
+type AwsIAM struct {
+	// RoleName
```

Comment:
the structs need to have comments since they are being exported but the struct fields themselves don't need comments unless there is something to mention; repeating the name doesn't really add anything - maybe just remove these?